### PR TITLE
GAP-1819: Submission Summary

### PR DESCRIPTION
**Pull request by ConorFayleTCO**
## Description

[GAP-1819](https://technologyprogramme.atlassian.net/browse/GAP-1819)

Adding summary page between submission sections page and submission submit.

I've added placeholder links for Change and download but they're properly implemented in https://github.com/cabinetoffice/gap-find-apply-web/pull/234. Also added some work for viewing completed submission but not hooked up.

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:
Video in ticket

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-1819]: https://technologyprogramme.atlassian.net/browse/GAP-1819?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      ) : (
+        <dd
+          className="govuk-summary-list__value"
+          id={response}
+          data-cy={`cy-section-value-${response}`}
+        >
+          {response ? response : '-'}
+        </dd>
+      )}
+      {readOnly ? null : (
```
Comment (jgunnCO) [inline packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx:258]:
```suggestion
      {!readOnly && (
```

File: packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    })
+  );
+
+  const mandatoryQuestionService = GrantMandatoryQuestionService.getInstance();
+  const mandatoryQuestionDto =
+    await mandatoryQuestionService.getMandatoryQuestionBySubmissionId(
+      submissionId,
+      jwt
+    );
+  const mandatoryQuestionId = mandatoryQuestionDto.id;
```
Comment (jgunnCO) [inline packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx:83]: Unless I'm missing something we pass this `mandatoryQuestionId` down to `QuestionRow` but we never actually do anything with it?

Comment (ConorFayleTCO) [reply packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx:83]: we will be as part of GAP-2096: https://github.com/cabinetoffice/gap-find-apply-web/pull/234/files#diff-541358a63993057c50641472fe8160e1e0ab9f51b9c1427200657f189c4c20e0R275

I was working on them in tandem so there's a bit of overlap between the two. I'm happy to pull this out here if you want though

Comment (jgunnCO) [reply packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx:83]: Ahh okay no worries if it's gonna be used for something.

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**